runtime/server/server.go

```diff
@@ -168,52 +174,22 @@ func (s *Server) ServeHTTP(ctx context.Context, registerAdditionalHandlers func(
 }
 
 // HTTPHandler HTTP handler serving REST gateway.
-func (s *Server) HTTPHandler(ctx context.Context, registerAdditionalHandlers func(mux *http.ServeMux)) (http.Handler, error) {
-	// Create REST gateway
-	gwMux := gateway.NewServeMux(
-		gateway.WithErrorHandler(HTTPErrorHandler),
-		gateway.WithOutgoingHeaderMatcher(func(s string) (string, bool) {
-			// grpc gateway adds gateway.MetadataHeaderPrefix to all outgoing headers
-			// we want to skip that for `x-trace-id` set in response
-			if s == observability.TracingHeader {
-				return s, true
-			}
-			// default matcher logic
-			return fmt.Sprintf("%s%s", gateway.MetadataHeaderPrefix, s), true
-		}),
-	)
-	opts := []grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())}
-	grpcAddress := fmt.Sprintf("localhost:%d", s.opts.GRPCPort)
-	err := runtimev1.RegisterRuntimeServiceHandlerFromEndpoint(ctx, gwMux, grpcAddress, opts)
-	if err != nil {
-		return nil, err
-	}
-	err = runtimev1.RegisterQueryServiceHandlerFromEndpoint(ctx, gwMux, grpcAddress, opts)
-	if err != nil {
-		return nil, err
-	}
-	err = runtimev1.RegisterConnectorServiceHandlerFromEndpoint(ctx, gwMux, grpcAddress, opts)
-	if err != nil {
-		return nil, err
-	}
+func (s *Server) HTTPHandler(ctx context.Context, registerAdditionalHandlers func(mux *http.ServeMux), transcoder *vanguard.Transcoder, local bool) (http.Handler, error) {
+	httpMux := http.NewServeMux()
 
-	// One-off REST-only path for multipart file upload
-	// NOTE: It's local only and we should deprecate it in favor of a cloud-friendly alternative.
-	err = gwMux.HandlePath("POST", "/v1/instances/{instance_id}/files/upload/-/{path=**}", auth.GatewayMiddleware(s.aud, s.UploadMultipartFile))
-	if err != nil {
-		panic(err)
+	// Register the Vanguard handler for gRPC transcoding
+	if local {
+		httpMux.Handle("/v1/", transcoder)
+	} else {
+		httpMux.Handle("/", transcoder)
```

Comment:
Does this mean the gRPC endpoints won't route correctly on local? Won't that break e.g. `rill project refresh --local`?

Consider mounting the explicit gRPC paths on root instead. Like:
```go
httpMux.Handle("/v1/", transcoder)
httpMux.Handle("/rill.runtime.RuntimeService/", transcoder)
httpMux.Handle("/rill.runtime.QueryService/", transcoder)
httpMux.Handle("/rill.runtime.ConnectorService/", transcoder)
```

Btw, I'm not sure those are the correct paths, so please double check :)